src/Managed.sol

```diff
@@ -3,23 +3,49 @@ pragma solidity ^0.8.17;
 
 import {Upgradeable} from "./Upgradeable.sol";
 import {StorageSlot} from "openzeppelin-contracts/utils/StorageSlot.sol";
+import {EnumerableSet} from "openzeppelin-contracts/utils/structs/EnumerableSet.sol";
+
+using EnumerableSet for EnumerableSet.AddressSet;
```

Comment:
What is the mean reason for the EnumerableSet?  Only to list all pausers with `allPausers()` I assume.

I think it is okay to use it, maybe a bit overengineered. Since, we rely on so many things just on events. 

However, we could argue that these roles come with great power, so it seems good. :+1:

Reply:
Yes, that's the reason. With `allPausers` you can even see on-chain who's a pauser, and it's trivial to query for them off-chain, it's a press of a button on Etherscan. When relying on events, it's more difficult and you don't get this information on-chain. The gas cost is larger, because instead of updating 1 slot you need to update 3, but IMO that's a fair trade off, pausers are not supposed to be updated often.